packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -979,7 +981,11 @@ function convertN8nRequestToAxios(n8nRequest: IHttpRequestOptions): AxiosRequest
 		// if there is a body and it's empty (does not have properties),
 		// make sure not to send anything in it as some services fail when
 		// sending GET request with empty body.
-		if (typeof body === 'string' || (typeof body === 'object' && !isObjectEmpty(body))) {
+		if (
+			typeof body === 'string' ||
+			(typeof body === 'object' &&
+				!(isObjectEmpty(body) && NoBodyHttpMethods.includes(method || 'GET')))
+		) {
```

Comment:
@feelgood-interface why is this change needed? can we avoid modifying this file?

Reply:
@michael-radency the update operations have 2 separate requests, depending on which fields are updated. To make the code cleaner I used the `routing.send` property for the first request. Microsoft may change the API to support all fields being updated in a single request. 
So currently it can happen that the first request has an empty body. This is not efficient, but this will happen rarely. 
The `removeEmptyBody` [httpRequest](https://github.com/n8n-io/n8n/blob/118fbb2139a17ef5d279b4efa81d2628267f3b7d/packages/core/src/NodeExecuteFunctions.ts#L1030C1-L1035C62) is contradicting the empty body check in `convertN8nRequestToAxios`. The one checks for the method type, the other one doesn't. In my opinion this should be consistent, i.e. checking the request method. 
```
export const removeEmptyBody = (requestOptions: IHttpRequestOptions | IRequestOptions) => {
	const method = requestOptions.method || 'GET';
	if (NoBodyHttpMethods.includes(method) && isEmpty(requestOptions.body)) {
		delete requestOptions.body;
	}
};
```
```
if (typeof body === 'string' || (typeof body === 'object' && !isObjectEmpty(body))) {
	axiosRequest.data = body;
}
```
However, if this is the intended behaviour, I can change the Entra node and manually check if the body is empty and then ignore the request.